src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DriverDrawerAdapter.java

```diff
@@ -5,16 +5,23 @@
 import edu.kis.legacy.drawer.shape.LineFactory;
 import edu.kis.powp.jobs2d.Job2dDriver;
 
+import javax.swing.*;
+
 /**
  * driver adapter to drawer with several bugs.
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class DriverDrawerAdapter extends DrawPanelController implements Job2dDriver {
 	private int startX = 0, startY = 0;
 
-	public MyAdapter() {
+	public DriverDrawerAdapter() {
 		super();
 	}
 
+	public DriverDrawerAdapter(JPanel panel) {
+		super();
+		this.initialize(panel);
```

Comment:
solution inconsistent with the instructions: Dostęp do instancji DrawPanelController zintegrowanej z oknem aplikacji jest zdefiniowany w metodzie getDrawerController() w klasie DrawerFeature.